### PR TITLE
761 inconsistent time imputation

### PR DESCRIPTION
**Pull request by alanaharris22**
…ert_dtc_to_dtm and derive_vars_dt

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by alanaharris22** [comment]
@pharmaverse/admiral-devs - pull request is ready for review.
Thanks, 
Alana

**Comment by alanaharris22** [comment]
@pharmaverse/admiral-devs - pull request ready for review. 
Thanks!

**Comment by bms63** [comment]
Hi Alanah can you investigate why the github actions are failing?  How come the .renv.loc file is showing a change?

**Comment by alanaharris22** [comment]
Hi @bms63, 

Thanks for your help! 

I think it might have been because I used `renv_snapshot()`? Anyway, all seems to be working fine now (last 2 commits were just me adding and removing a test comment). 

I did just notcie a new comment on #727 point 11 (changing default to `"auto"` rather than `"none"` in `derive_vars_dt()`) so I think I might need to make some more updates. 

All the best, 
Alana

**Comment by bms63** [comment]
Good to hear!  Let us know when ready to review.  You can use @pharmaverse/admiral tag to ask people to review.

**Review by bms63 — APPROVED** [review]
Do we need some type of warning message bringing this to the attention of users?  For example, if they already have this in there code and update to 0.7.0 and it falls over - should we tell this has been changed in 0.7.0.  Or does one of the assert_ statements catch this for us?  Everything else looks good to me.

**Comment by alanaharris22** [comment]
Hi @bms63, for `derive_vars_dt()`, `assert_character_scalar()` will give a warning if `flag_imputation` isn't one of "none", "auto", or "date" - do you think that is enough?

**Comment by bms63** [comment]
> Hi @bms63, thanks for reviewing! For `derive_vars_dt()`, `assert_character_scalar()` will give a warning if `flag_imputation` isn't one of "none", "auto", or "date" - do you think that's enough?

Sounds good to me!

**Comment by rossfarrugia** [comment]
@alanaharris22 would you have time to update the merge conflicts on this one today or tomorrow so that it could be merged ahead of the release end of this week? let me know if you need help

**Comment by alanaharris22** [comment]
Hi @rossfarrugia - the merge conflicts should all be okay now. I did notice the Roxygen and Style checks are failing - should I look into these too? 
Thanks!

**Comment by bms63** [comment]
I'm still trying to understand what this output means for styler and what we need to do about it.  

Is it changing some files for us and then for others that it detects changes that have too many style change telling us to do it manually?

If those files are changed by the action, do I just pull down to get those updates and then those styling issues won't show up anymore?  Maybe time for some additional guidance on this! :)  Who wants to make an issue!?!

Can you play around with this @ClaudiaCarlucci and let us know if I am right or wrong.  

I think we ignore the roxygen fail for now.

**Comment by thomas-neitmann** [comment]
@dinakar29 Can you reply to @bms63's question above?

**Comment by alanaharris22** [comment]
Hello @bms63, 
The styler seemed to want to change 4 files, I ran `styler::style_file(c("R/derive_date_vars.R", "tests/testthat/test-derive_merged.R", "vignettes/bds_finding.Rmd", "vignettes/imputation.Rmd"))` which was what it suggested in the check window. Looking at the code it has just made a load of formatting changes/indenting etc. The check is passing now too! 
Thanks!

**Comment by bundfussr** [comment]
I think in most cases the changes made by `style_file()` are OK. I had two scenarios where I did not accepted the changes:

- If the columns in a `tribble()` call are not left or right aligned, `style_file()` removes the alignment. In this cases I have updated the call to avoid that the alignment is removed, i.e., I left or right aligned all columns.
- Sometimes the indentation of function calls is changed such that the arguments are at different levels. For example
    ```
      actual <- derive_vars_merged(advs,
                                   dataset_add = adsl,
                                   by_vars = vars(STUDYID, USUBJID))
    ```
    is changed to
    ```
      actual <- derive_vars_merged(advs,
        dataset_add = adsl,
        by_vars = vars(STUDYID, USUBJID))
    ```
    I changed such calls to
    ```
      actual <- derive_vars_merged(
        advs,
        dataset_add = adsl,
        by_vars = vars(STUDYID, USUBJID))
    ```

**Comment by dinakar29** [comment]
@bundfussr - You can actually [customize the style](https://styler.r-lib.org/articles/customizing_styler.html) or [remove rules](https://styler.r-lib.org/articles/remove_rules.html) per your preferences, this will set a common style guide for all contributors and enforce it via CI.

Thanks @alanaharris22 for the guidance  - that is exactly righjt.
@bms63, if you look at the [log for ROxygen](https://github.com/pharmaverse/admiral/runs/6467323950?check_suite_focus=true), similar to the styler output, you will see remediation steps to resolve the issues with missing man pages.

tl;dr: Run this in your session with Roxygen 7.2.0 installed:

```R
roxygen2::roxygenize('.', roclets = c('rd', 'collate', 'namespace'))
```

And commit + push the modified man pages to this branch/PR.

**Comment by alanaharris22** [comment]
Hi @dinakar29, @bms63 - please see latest updates after running `roxygen2::roxygenize('.', roclets = c('rd', 'collate', 'namespace'))`
Thanks!

**Review by bms63 — APPROVED** [review]

**Comment by bms63** [comment]
Looks like the Templates have fallen over.  I was poking around and see this doesn't have latest updates from devel.  Maybe that is the cause?

**Review by millerg23 — APPROVED** [review]
Have approved this again, to see if this triggers the templates to run.

**Comment by bms63** [comment]
Some issue with the templates - saw something strange in the adpp template and pushed up a change

**Comment by bms63** [comment]
@millerg23 can you re-approve to see if check templates trigger

**Review by millerg23 — APPROVED** [review]
Have approved this again, to see if this triggers the templates to run.

**Comment by bms63** [comment]
@millerg23 need another approval think i found the issue

File: inst/templates/ad_adlb.R
Comment (bundfussr) [inline inst/templates/ad_adlb.R:93]: I think this line should be removed such that the default (`"auto"`) is used. This ensures compliance with ADaM IG.

**Comment by bms63** [comment]
@bundfussr or @millerg23 can you re-approve so examples fire off again?

**Comment by bms63** [comment]
@alanaharris22 we got it!  thanks again for your hard work here!!

**Comment by alanaharris22** [comment]
Thank you!! 
Sorry I wasn't much help there at the end!

**Comment by rossfarrugia** [comment]
@alanaharris22 you were a massive help! thanks for your patience here with this PR as we had lots of new checks getting set-up. this was a tricky one to pick up too as one of your first issues as there were lots of knock-on impacts - nice job!

**Pull request merged**
